File: caffeine/src/main/java/project/study/caffeine/core/domain/city/entity/City.java
package project.study.caffeine.core.domain.city.entity;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Objects;

@Entity(name = "city")
public class City {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long cityId;

    @Column
    private String name;

    public Long getCityId() {
        return cityId;
    }

    public String getName() {
        return name;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof City city)) return false;
        return getCityId().equals(city.getCityId());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getCityId());
    }

    @Override
    public String toString() {
        return cityId.toString();
    }
}

File: jpa/src/main/kotlin/project/study/jpa/core/domain/city/entity/City.kt
package project.study.jpa.core.domain.city.entity

import jakarta.persistence.*

@Entity
class City(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var cityId: Long? = null,

    @Column
    private var name: String
) {
    fun getCityId(): Long? {
        return cityId
    }

    fun getName(): String {
        return name
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is City) return false
        if (cityId != other.cityId) return false
        return true
    }

    override fun hashCode(): Int {
        return cityId?.hashCode() ?: 0
    }

    override fun toString(): String {
        return cityId.toString()
    }
}
